### PR TITLE
Windows CI

### PR DESCRIPTION
**Pull request by pfilipko1**
## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: .github/workflows/build-on-windows.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  build-executable:
+   runs-on: windows-latest
+   steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: install chocolatey
```
Comment (marcin-ol) [inline .github/workflows/build-on-windows.yml:19]: Shouldn't be necessary to install `chocolatey`, try removing.
According to documentation on Github `choco` [should already be present on workers running windows-latest](https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md#package-management).

File: .github/workflows/build-on-windows.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        run: Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+      
+      - name: install wget
```
Comment (marcin-ol) [inline .github/workflows/build-on-windows.yml:22]: Shouldn't need to install `wget`, github workers should already support `curl` [as it comes with Windows](https://learn.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows).
Use `curl` instead.

File: scripts/windows/build-pyspy.bat [hunk 3]
```diff
@@ -14,7 +14,7 @@ git clone --depth 1 -b %TAG% https://github.com/Granulate/py-spy.git && git -C p
 
 IF NOT EXIST .\dep\vs_BuildTools.exe (
 	ECHO Downloading Visual Studio Build Tools...
-	wget -O .\dep\vs_BuildTools.exe https://aka.ms/vs/17/release/vs_buildtools.exe
+	curl -o .\dep\vs_BuildTools.exe https://aka.ms/vs/17/release/vs_buildtools.exe
```
Comment (marcin-ol) [inline scripts/windows/build-pyspy.bat:17]:
```suggestion
	curl -sfLo .\dep\vs_BuildTools.exe https://aka.ms/vs/17/release/vs_buildtools.exe
```

File: scripts/windows/build-pyspy.bat [hunk 4]
```diff
@@ -43,7 +43,7 @@ SET ERRORLEVEL=
 WHERE cargo
 IF ERRORLEVEL 1 (
 	ECHO cargo wasn't found. Attempting to install...
-	wget -O .\dep\rustup-init.exe https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe
+	curl -o .\dep\rustup-init.exe https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe
```
Comment (marcin-ol) [inline scripts/windows/build-pyspy.bat:46]:
```suggestion
	curl -sfLo .\dep\rustup-init.exe https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe
```

File: scripts/windows/build.bat [hunk 5]
```diff
@@ -39,7 +33,7 @@ SET ERRORLEVEL=
 WHERE /Q pip
 IF ERRORLEVEL 1 (
         ECHO pip wasn't found. Attempting to install...
-        wget -O .\dep\get-pip.py https://bootstrap.pypa.io/get-pip.py
+        curl -o .\dep\get-pip.py https://bootstrap.pypa.io/get-pip.py
```
Comment (marcin-ol) [inline scripts/windows/build.bat:36]:
```suggestion
        curl -sfLo .\dep\get-pip.py https://bootstrap.pypa.io/get-pip.py
```

**Comment by marcin-ol** [comment]
One more thing.
I suggest to remove
```
SET ERRORLEVEL=
```
in all Windows build file, **unless** used before calling local routines (calls to labels); our bat files don't do it right now.

Rationale:
- condition `IF ERRORLEVEL` doesn't check the environment variable `%ERRORLEVEL%`; see [this old blog "ERRORLEVEL is not %ERRORLEVEL%"](https://devblogs.microsoft.com/oldnewthing/20080926-00/?p=20743);
- resetting `ERRORLEVEL` should not be needed before calling commands like `python`, `WHERE`, `wget` as well as built-in commands (`dir`) and so on; they set it by default.

**Review by marcin-ol — APPROVED** [review]
LGTM

**Comment by Jongy** [comment]
Failrues are not related - I mege it.

@pfilipko1 it can be nice to add storing the artifact, so it can be downloaded from the CI. Let's add in a separate PRs?

**Review by Jongy — APPROVED** [review]

**Pull request merged**